app/templates/components/conversations/conversation-thread.hbs

```diff
@@ -1,37 +1,31 @@
 <div class="conversation-thread-inner">
   <div class="conversation-thread__messages">
     {{#if conversation.message.isLoaded}}
-      {{conversations/conversation-part-comment
-        author=conversation.message.author
-        body=conversation.message.body
-        readAt=conversation.message.readAt
-        sentAt=conversation.message.insertedAt
-      }}
-    {{/if}}
-    {{#each sortedConversationParts as |conversationPart|}}
-      {{#if conversationPart.isLoaded}}
-        {{#if (eq conversationPart.partType 'closed')}}
-          {{conversation/conversation-part-closed
-            author=conversationPart.author
-            closedAt=conversationPart.insertedAt
-          }}
-        {{else if (eq conversationPart.partType 'reopened')}}
-          {{conversations/conversation-part-comment
-            author=conversationPart.author
-            body=conversationPart.body
-            readAt=conversationPart.readAt
-            sentAt=conversationPart.insertedAt
-          }}
-        {{else}}
-          {{conversations/conversation-part-comment
-            author=conversationPart.author
-            body=conversationPart.body
-            readAt=conversationPart.readAt
-            sentAt=conversationPart.insertedAt
-          }}
+      {{#each sortedConversationParts as |conversationPart|}}
+        {{#if conversationPart.isLoaded}}
+          {{#if (can "administer project" project)}}
+            {{#if (eq conversationPart.partType 'closed')}}
+              {{conversation/conversation-part-closed
+                author=conversationPart.author
+                closedAt=conversationPart.insertedAt
+              }}
+              {{else if (eq conversationPart.partType 'reopened')}}
+                {{conversations/conversation-part-reopened
+                author=conversationPart.author
+                reopenedAt=conversationPart.inserted
+              }}
+            {{/if}}
+          {{else}}
+            {{conversations/conversation-part-comment
+              author=conversationPart.author
+              body=conversationPart.body
+              readAt=conversationPart.readAt
+              sentAt=conversationPart.insertedAt
+            }}
+          {{/if}}
         {{/if}}
-      {{/if}}
-    {{/each}}
+      {{/each}}
+    {{/if}}
   </div>
   <div class="conversation-thread__composer">
     {{conversations/conversation-composer
```

Comment:
Your logic here is flawed. A user who cannot administer will simply be shown all of the conversation parts, regardless of type, except they will be displayed as if they are comments.

You lack a test that would check for this, so you should add that to

```Javascript
test("only comment parts are shown if user cannot administer")
```

There are multiple ways you could achieve this. The most straightforward would be to rearrange the logic here so it just works, but it would probably make for a complicated and hard to follow template file. Try to think your way through this. 

The aim is, if I have a conversation which was commented on twice, then was closed, then reopened

- a user who can administer the project will see a total of 5 parts. 1 for the original message body, 2 comments, 1 closed, 1 reopened
- a user who cannot administer the project will see a total of 3 parts, 1 for the original message body, 2 for comments

Your test should support that goal. The tests you modified are fine. What you're missing are tests for behavior when user is not an admin.